core/spec/models/spree/base_spec.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Spree::Base do
+  before(:all) do
+    class CreatePrefTest < ActiveRecord::Migration[4.2]
+      def self.up
+        create_table :pref_tests do |item|
+          item.string :col
+          item.text :preferences
+        end
+      end
+
+      def self.down
+        drop_table :pref_tests
+      end
+    end
+
+    @migration_verbosity = ActiveRecord::Migration[4.2].verbose
+    ActiveRecord::Migration[4.2].verbose = false
+    CreatePrefTest.migrate(:up)
+  end
+
+  after(:all) do
+    CreatePrefTest.migrate(:down)
+    ActiveRecord::Migration[4.2].verbose = @migration_verbosity
+  end
+
+  context "with a class that has a preference column but does not explicitly serialize" do
+    before :all do
+      class PrefTestWithoutSerialization < Spree::Base
+        self.table_name = :pref_tests
+      end
+    end
+
+    before(:each) do
+      allow(Spree::Deprecation).to receive(:warn).
```

Comment:
What about turning this into an `expect`. The concern is that it may not be emitting the deprecation warning when we expect this happens. I mean, stubbing the deprecation should not be used to avoid failures and clean the logs only but to actually test that users get a deprecation message.

Reply:
Got it, and I can't think at a clean way to do it.